src/backend/cpu/copy.hpp

```diff
@@ -38,6 +38,7 @@ namespace cpu
                              const dim4& upperBoundPadding,
                              const af::borderType btype)
     {
+        in.eval();
```

Comment:
why is this necessary?

Reply:
CPU backend implementations eval inputs prior to enqueue of their respective algorithm. I am not sure whats the fundamental criteria to that ? Followed that convention

Reply:
you don't need to do that. Did you run into any issues if you didn't do this?